### ISSUE TITLE
Sign up: Domain First: Allow domain connection to existing sites

### ISSUE THREAD

**Issue by mikeshelton1503** [Automattic/wp-calypso#12196]
We are allowing existing users to access to wordpress.com/domains however currently their is no option to add the domain to an existing site.

We need to add the option:

![image](https://cloud.githubusercontent.com/assets/12596797/23968500/504de30c-0999-11e7-83d4-3b2334503d64.png)

I struggled a bit with the image for the existing site card. The other two are simplified page layouts which makes it difficult to have an image that's not a page layout not look inconsistent. I'm open to ideas for that image. cc @danhauk @breezyskies 

Also, I just realized a possible confusion with the copy I used. "Add to an existing site" could be interrupted as a way to add to a non-WordPress.com site. cc @ranh for thoughts on the copy.

Upon selecting "Add to an existing site" we need to show the site picker:

![image](https://cloud.githubusercontent.com/assets/12596797/23968608/d23b9e04-0999-11e7-86e3-019a34bbf6f0.png)

**Comment by drw158** [comment]
Hm, yea that is tough. That flat rectangle style of illustration isn't very flexible. The "Just buy a domain" one is pretty silly. The image you created doesn't really work. I think you'd have to either update the other images to fit the style, or tweak your image to fit within the existing one.

I think you could simplify your image to conform. Maybe just the browser window with a plus? No dashes or additional colors.

Would be great to add more detail to the other images though if you have time.

**Comment by mikeshelton1503** [comment]
Thanks @drw158 I agree ideally we would recreate the other images however those are based off of the others used in NUX so I think go the route of simplification for now.

**Comment by mikeshelton1503** [comment]
Ok I simplified the browser, brought in some of the greeking from the other images and changed the icon color to dark blue.

![image](https://cloud.githubusercontent.com/assets/12596797/23978236/4a709e18-09c9-11e7-898e-a423f95d800e.png)

**Comment by ranh** [comment]
> Also, I just realized a possible confusion with the copy I used. "Add to an existing site" could be interrupted as a way to add to a non-WordPress.com site. cc @ranh for thoughts on the copy.

Good point, but there's an opportunity here. Why should "Add to an existing site" be limited to WordPress.com sites only? 

On the "Choose your site" screen in your mockup, we can add an option "Add a site not hosted with WordPress.com". This could be as involved as a full-fledged flow to connect the domain to third-party hosting providers, or as simple as linking to some version of the "Manage Domain" screen or even just a support doc. 

----

Otherwise, a simple fix would be to change the labels to "Start a new WordPress.com site" and "Add to an existing WordPress.com site". 

But I think we should also present these options differently. "Start a new site" and "Add to an existing site" should be grouped together, and separated from the "Just buy a domain" option. 

For example, we can show a choice between "Just a domain" and "WordPress.com site", and then for "WordPress.com site" show a another set of choices for "new site" and "existing site".

Also, this idea may not work with this design, but another way to clarify that "existing site" means only a site on WordPress.com is to have the available sites (or a few of the available sites) already visible before making the choice. For example, in a dropdown or some minimal site-selector that is disabled until the option is selected.

**Comment by folletto** [comment]
> Upon selecting "Add to an existing site" we need to show the site picker:

Can we should the list already opened? That would make all the sites already visible, and avoids an extra click to just open the list :)

**Comment by mikeshelton1503** [comment]
>Good point, but there's an opportunity here. Why should "Add to an existing site" be limited to WordPress.com sites only?

You're right, there's a good opportunity there. However, I think that we should leave that as a separate enhancement. It adds a bit more complexity (development wise, not for the user) because the plan for now was to only show this option if a user is logged in.

>I think we should also present these options differently. "Start a new site" and "Add to an existing site" should be grouped together, and separated from the "Just buy a domain" option.

Good idea, I'll try that out!

**Comment by mikeshelton1503** [comment]
>Can we should the list already opened? That would make all the sites already visible, and avoids an extra click to just open the list :)

👍 Good call.

**Comment by drw158** [comment]
new image looks much more consistent I think 👍

**Comment by mikeshelton1503** [comment]
Ok I've made several changes to address concerns about the illustration and copy:

![image](https://cloud.githubusercontent.com/assets/12596797/24005566/be920a7c-0a3f-11e7-95bd-c2e1abe2f327.png)

**What changed:**
 - The illustrations are now more consistent with one another while remaining in the simplistic style of the other NUX illustrations.
- Removed the subtitle copy since it wasn't accurate and wasn't adding value. Instead @ranh and I decided to just use the headline to communicate the instruction.
- Clarified that these options are only for creating or adding to WordPress.com sites (atm)
- Reorganized order of options to group more naturally. (ht @ranh)

**Comment by josemarques** [comment]
Any reason why the "just buy a domain" option needs a different background color? To me it feels a bit out of balance due to it. 
Domains are hard to illustrate and I don't have a better example for that particular spot but I'm not reading anything from that illustration :(. In that context it looks like the window bar that is present in the other two illustrations. We use the globe on the domains link in the sidebar, maybe play a bit with that?

**Comment by mikeshelton1503** [comment]
>Any reason why the "just buy a domain" option needs a different background color? To me it feels a bit out of balance due to it.

The idea was that since "just buy a domain" is not like the other options, which involve actively using the domain, we would separate it. I originally tried a vertical line a separator but that didn't work too well, then arrive at the color difference which may not be perfect either.

>Domains are hard to illustrate and I don't have a better example for that particular spot but I'm not reading anything from that illustration :(.

Yea totally agree, that's the illustration is currently being used. Originally I hadn't intended to change the existing illustrations with this change, however for consistency sake I did change the "Start a new site" one. We could also revisit the "just buy a domain one" but I might opt to do that outside of the issue because, well I don't have any great ideas for improving it atm.

**Comment by danhauk** [comment]
I agree with that the domain option looks out of place here. I think since the other two illustrations aren't exactly visual representations of UI we can branch out a bit on the illustration for the "just buy a domain". I played around with using the top of the browser window as the illustration. It's not perfect, but it's better.

![choose-domain-option](https://cloud.githubusercontent.com/assets/448298/24055438/6684c8ba-0b16-11e7-886d-20b8b81407e1.png)

**Comment by mikeshelton1503** [comment]
Ooh, nice! Thanks Dan!

It's much more consistent and also a great improvement on clarity.

**Comment by mikeshelton1503** [comment]
Ok I think this is a good place to start implementing. For ease of reading I'll summarize the change to be made:

**If a user is logged in, provide option to add domain to new site**

![image](https://cloud.githubusercontent.com/assets/12596797/24056074/d3be6c7c-0b18-11e7-944e-771a033757ac.png)

**If a logged in user chooses "Add to my existing WordPress.com site", show the site picker**

![image](https://cloud.githubusercontent.com/assets/12596797/24056309/92a74654-0b19-11e7-8602-2ff194504f12.png)

**Comment by mikeshelton1503** [comment]
@aidvu The design for this is ready. Since it's a new flow, I think it's best if Cobalt tackles this one.

I can provide the svgs for the new illustrations, just let me know whether that's best done in a separate PR or as part of the PR that adds this flow.

**Comment by aidvu** [comment]
Can do it in the same PR when we get there. :) Thanks @mikeshelton1503!

**Comment by ranh** [comment]
@mikeshelton1503, as mentioned on Slack, what do you think about aligning the top bar in all three illustrations? The might make the domain-only option look more like an "empty" site.

For example:

![top](https://cloud.githubusercontent.com/assets/426518/24080323/3be07f10-0ca5-11e7-9ac8-13cba3784077.png)


cc @danhauk @josemarques

**Comment by ranh** [comment]
Jumping on this issue with a related copy tweak. 

@breezyskies mentioned that the label "just buy a domain" might lead to some confusion, since we later show the domain under My Sites in Calypso. 

Let's try this instead:

> Start a new site now.

> Add to my existing WordPress.com site.

> Show a "Coming soon" page (you can add a site later).

Other than avoiding any confusion about My Sites, this also offers a better description of what is going to happen, reiterates that a site can be added later (we already say this on the front page, but here is where it's relevant), and avoids any technical jargon like "domain", "landing page", or "temporary site".

**Comment by mikeshelton1503** [comment]
It was recently brought to our attention that we should consider the use case of existing Jetpack sites. Ideally we just allow you to connect to a JP site, however that's not possible without some significant development. So lets start with something more manageable for now:

If a user has any Jetpack sites display an info notice above the list that says:

>Some of your sites aren't listed because we can't add the domain automatically. You can **purchase the domain** only and then add your site manually.

cc @ranh for copy review

**Comment by ranh** [comment]
@mikeshelton1503 Sorry for the delayed review. +1 for adding the Jetpack notice. I'm not sure the second part is necessary, since the directions are vague and won't be very helpful. And we don't have room for much else.

Something like this might work:

> Some of your sites aren't listed because we can't add the domain automatically. Check with your host about adding a domain to your site.

I think this is more likely to lead some users to good results. However I'm not familiar enough with Jetpack to make a good judgement call here.

Perhaps it would be better to leave the second part out entirely. Or replace it with a "learn more" (or similar) link to a support page for connecting domains registered on WordPress.com to Jetpack sites.

**Comment by ranh** [comment]
Noting that I updated the copy in [my comment above](https://github.com/Automattic/wp-calypso/issues/12196#issuecomment-289515842). The label for the last option ("domain only"):

> Show a "Coming soon" message (add a full site later).

Previously it said "add a site later", which defeats the original purpose of the edit by again suggesting to users that they are not getting a site at all. 

cc @breezyskies since we talked about that.
